src/BloomExe/Publish/PdfViewer.cs

```diff
@@ -99,9 +99,13 @@ public bool ShowPdf(string pdfFile)
 				// and continue to show it using that.
 			}
 #endif
-
-			var url = string.Format("{0}{1}?file=/bloom/{2}", Bloom.web.ServerBase.PathEndingInSlash,
-				FileLocator.GetFileDistributedWithApplication("pdf/web/viewer.html"), pdfFile);
+			// Escaping the filename twice for characters like # is needed in order to get the
+			// pdf filename reliably through to our local server.  This is to prevent the filename
+			// from being cut short at the # character.  See https://jira.sil.org/browse/BL-951.
+			var url = string.Format("{0}{1}?file=/bloom/{2}",
+				Bloom.web.ServerBase.PathEndingInSlash,
+				FileLocator.GetFileDistributedWithApplication("pdf/web/viewer.html"),
+				pdfFile.EscapeCharsForHttp().EscapeCharsForHttp());
```

Comment:
This URL is being sent to gecko/xul.  It strips off one level of escaping immediately as far as I can tell, then truncates the query portion at a #, assuming that it must be an internal marker.  The second level of escaping is then removed as the localhost retrieves the remaining url.  I could repeat my experiments, as this is something that was empirically discovered while trying to figure out how to get book names with # in them to display as PDF files in the publish pane.